### PR TITLE
Try to update comments for flakes instead of making new ones.

### PR DESCRIPTION
**Pull request by rmmh**
Conditions where it will update a comment:
- It is referring to a flake with the "same reason" (fuzzy equality)
- The comment is less than 2 weeks old.
- Nobody other than the bot has made a comment after that.

This should cut down on a large volume of flakes comments and respect
people's inboxes much more.

### CODE REVIEW

**Comment by foxish** [comment]
@grodrigues3, it appears that it is suggesting an OWNER from the top-level OWNERS file here. bprashanth is not an owner listed in the mungegithub directory.

File: mungegithub/mungers/flake-manager.go [hunk 1]
```diff
@@ -226,6 +235,78 @@ func (p *individualFlakeSource) Body(newIssue bool) string {
...
+	a = flakeReasonOrdinalRE.ReplaceAllStringFunc(a, repl)
+
+	matches = make(map[string]string) // reset mapping
+	b = flakeReasonOrdinalRE.ReplaceAllStringFunc(b, repl)
+
+	if len(a) < 100 {
+		return a == b
+	}
+
+	// glog.Errorf("EQUIVALENCE AFTER REMAPPING:\n\n`%s`\n\n`%s`", a, b)
```
Comment (spxtr) [inline mungegithub/mungers/flake-manager.go]: ?

File: mungegithub/mungers/flake-manager.go [hunk 2]
```diff
@@ -41,7 +43,14 @@ const failedStr = "Failed: "
 var (
 	// pullRE is a regexp that will extract the PR# from a path to a flake
 	// that happened on a PR.
-	pullRE = regexp.MustCompile("pull/([0-9]+)/")
+	pullRE               = regexp.MustCompile("pull/([0-9]+)/")
+	failureInnerRE       = regexp.MustCompile("(?s)(.*)\n" + failedStr + "([^\n]*)\n\n```\n(.*)\n```\\s*$")
```
Comment (spxtr) [inline mungegithub/mungers/flake-manager.go]: Please comment saying what these are supposed to match. `gubernatorLinkRE` is clear enough, but why do we need the others?

File: mungegithub/mungers/flake-manager.go [hunk 3]
```diff
@@ -226,6 +235,78 @@ func (p *individualFlakeSource) Body(newIssue bool) string {
 	return body
 }
 
+func flakeReasonsAreEquivalent(a, b string) bool {
```
Comment (spxtr) [inline mungegithub/mungers/flake-manager.go]: Comment saying what this considers equivalent.

File: mungegithub/mungers/flake-manager_test.go [hunk 4]
```diff
@@ -190,3 +215,113 @@ func TestPullRE(t *testing.T) {
...
+/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:158`,
+			`/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:118
+Sep 29 22:41:21.062: Test failed; at least one node failed to reboot in the time given.
+/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/reboot.go:158`,
+		},
+	}
+	for i, test := range testcases {
+		result := flakeReasonsAreEquivalent(test.a, test.b)
+		if result != test.expected {
+			//t.Errorf("test #%d failed", i)
```
Comment (spxtr) [inline mungegithub/mungers/flake-manager_test.go]: ?

File: mungegithub/github/github.go [hunk 5]
```diff
@@ -1918,6 +1919,34 @@ func (obj *MungeObject) DeleteComment(comment *github.IssueComment) error {
 	return nil
 }
 
+// UpdatedComment will change the specified comment's body.
```
Comment (spxtr) [inline mungegithub/github/github.go]: `UpdateComment` :)

**Comment by spxtr** [comment]
/lgtm cancel

Forgot that it'll merge if I say that :P

**Comment by rmmh** [comment]
Added more comments, removed debugging lines.

**Pull request merged**